runtime/shared_common/shrinit.cpp

```diff
@@ -5186,4 +5274,36 @@ j9shr_getCacheDir(J9JavaVM* vm, const char* ctrlDirName, char* buffer, UDATA buf
 	return SH_OSCache::getCacheDir(vm, ctrlDirName, buffer, bufferSize, cacheType, false);
 }
 
+/**
+ * Determine existing top layer number
+ *
+ * @param [in] vm  The Java VM
+ * @param [in] ctrlDirName  The control dir name
+ * @param [in] cacheName  The cache name
+ * @param [in] runtimeFlags  The runtime flags
+ * @param [out] maxLayerNo  The existing top layer number
+ */
+static void
+findExistingCacheLayerNumbers(J9JavaVM* vm, const char* ctrlDirName, const char* cacheName, U_64 runtimeFlags, I_8 *maxLayerNo)
+{
+	I_8 maxLayer = -1;
+
+	J9PortShcVersion versionData;
+	setCurrentCacheVersion(vm, J2SE_VERSION(vm), &versionData);
+	versionData.cacheType = getCacheTypeFromRuntimeFlags(runtimeFlags);
+
+	for (I_8 l = 0; l <= J9SH_LAYER_NUM_MAX_VALUE; l++) {
```

Comment:
`l` is a particularly bad name for a variable - it looks too much like `1` in some fonts. Please use `layer` instead.

Reply:
Changed to use layer